homeassistant/components/volumio/media_player.py

```diff
@@ -260,4 +261,14 @@ async def async_browse_media(self, media_content_type=None, media_content_id=Non
         if media_content_type in [None, "library"]:
             return await browse_top_level(self._volumio)
 
-        return await browse_node(self._volumio, media_content_type, media_content_id)
+        return await browse_node(
+            self, self._volumio, media_content_type, media_content_id
+        )
+
+    async def async_get_browse_image(
+        self, media_content_type, media_content_id, media_image_id=None
+    ):
+        """Get album art from Volumio."""
+        cached_url = self.thumbnail_cache.get(media_content_id)
```

Comment:
Can we use `lru_cache` with a max size here instead to avoid leaking too much memory https://docs.python.org/3/library/functools.html

Reply:
The problem is that if we set `maxsize=x` and there are x+1 entries in the browser, media would not be displayed for 1 of those. I think a better approach might be to flush this cache on every new browse. That way the cache will only contain the result of the last query, and can't grow unbounded. What do you think?